virttest/utils_disk.py

```diff
@@ -601,6 +603,167 @@ def create_filesystem(session, partition_name, fstype, ostype, timeout=360):
         create_filesyetem_linux(session, partition_name, fstype, timeout)
 
 
+def get_mpoint_fstype_linux(session, partition):
+    """
+    Get a partition's file system type and mountpoint.
+
+    :param session: session object to guest.
+    :param partition: disk partition, like /dev/sdb1.
+    :return: Tuple (mountpoint, fstype)
+    """
+    mount_list = session.cmd_output('cat /proc/mounts')
+    mount_info = re.search(r'%s\s(.+?)\s(.+?)\s' % partition, mount_list)
+    if mount_info:
+        return mount_info.groups()
+    raise exceptions.TestError('partition is not mounted, mounted'
+                               'list as:\n%s' % mount_list)
+
+
+def get_partition_attrs_linux(session, partition):
+    """
+    Get partition attributes in linux guest.
+
+    :param session: session object to guest.
+    :param partition: like /dev/sdb1.
+    :return: dict like {'start': '512B', 'end': '16106127359B',
+                       'size': '16106126848B', 'type': 'primary'}
+    """
+    if "nvme" in partition:
+        pattern = r'(/dev/.*)p(\d+)'
+    else:
+        pattern = r'(/dev/.*)(\d+)'
```

Comment:
Better to `pattern = r'(/dev/.*)p(\d+)' if "nvme" in partition else r'(/dev/.*)(\d+)'` it's more pythonic  :)

Reply:
it's ok~